resolver/resolver_test.go

```diff
@@ -0,0 +1,118 @@
+package resolver
+
+import (
+	"net"
+	"net/http"
+	"testing"
+)
+
+func TestResolve8888(t *testing.T) {
+	r, err := NewResolver([]string{"8.8.8.8:53"})
+	if err != nil {
+		t.FailNow()
+	}
+
+	net.DefaultResolver = r
+
+	resp, err := http.Get("https://www.google.com/")
+
+	if err != nil {
+		t.Logf("error from http.Get(): %s", err)
+		t.FailNow()
+	}
+
+	resp.Body.Close()
+}
+
+func TestResolveAddresses(t *testing.T) {
+	table := map[string]struct {
+		input          []string
+		want           []string
+		expectError    bool
+		expectMismatch bool
```

Comment:
This test is testing `normalizeAddresses`, not address resolution as the name implies. With the reuse of the standard library function, I don't see much value in testing this. However, if we were to do it, testing for specific errors and addresses to be returned is the way to go (instead of these booleans).

Reply:
see above

Reply:
OK, cool, then rename the test name to `TestNormalizeAddresses` and structure the test table as:

```go
struct {
    in []string
    out []string
    err error
}
```

With the assertions being something like:

```go

out, err := normalizeAddrs(tc.in)
if got, want := out, tc.out; !reflect.DeepEqual(got, want) {
    t.Errorf("got addrs %v, want %v", got, want)
}

if got, want := fmt.Sprint(err), fmt.Sprint(tc.err); got != want {
    t.Errorf("got err %v, want %v", got, want)
}
```